File: 2024_java_basics/10-for-loops-lab/tasks/02-half-sum-element.java
import java.util.Scanner;

public class HalfSumElement {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int rows = scanner.nextInt();
        int sum = 0;
        int max = Integer.MIN_VALUE;

        for (int i = 0; i < rows; i++) {
            int num = scanner.nextInt();
            sum += num;

            if (num > max) {
                max = num;
            }
        }

        int sumNoMax = sum - max;

        if (max == sumNoMax) {
            System.out.println("Yes");
            System.out.println("Sum = " + max);
        } else {
            int diff = Math.abs(max - sumNoMax);
            System.out.println("No");
            System.out.println("Diff = " + diff);
        }
        
        scanner.close();
    }
}

File: 2022_c#_basics/10-for-loop-exercise/02-half-sum-element.cs
using System;

namespace _02_Half_Sum_Element
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int rows = int.Parse(Console.ReadLine());
            int sum = 0;
            int max = int.MinValue;

            for (int i = 0; i < rows; i++)
            {
                int num = int.Parse(Console.ReadLine());
                sum += num;

                if (num > max)
                {
                    max = num;
                }
            }

            int sumNoMax = sum - max;

            if (max == sumNoMax)
            {
                Console.WriteLine("Yes");
                Console.WriteLine("Sum = " + max);
            }
            else
            {
                int diff = Math.Abs(max - sumNoMax);
                Console.WriteLine("No");
                Console.WriteLine("Diff = " + diff);
            }
        }
    }
}
